pkg/ruler/service_discovery.go

```diff
@@ -22,6 +22,10 @@ const (
 	mechanismName = "dns_sd"
 )
 
+func init() {
+	discovery.RegisterConfig(dnsServiceDiscovery{})
```

Comment:
```suggestion
	discovery.RegisterConfig(&dnsServiceDiscovery{})
```

I think this should be a pointer to the struct as per the other sd structs registered in prometheus

Reply:
I don't think it's necessary since `dnsServiceDiscovery`'s methods don't take a pointer. If a pointer were necessary, it wouldn't have compiled.